skimage/feature/_kb.py

```diff
@@ -0,0 +1,264 @@
+
+    Examples
+    --------
+    >>> from skimage.feature import kb
+    >>> from skimage.color import rgb2gray
+    >>> from skimage.data import astronaut
+    >>> image = astronaut()[100:300, 100:300]
+    >>> kb(rgb2gray(image))
+    array([[   0.875,  102.625,   12.75 ],
+           [ 127.875,   72.125,   12.   ],
+           [   1.875,  146.125,   12.   ],
+           [ 163.875,  149.875,   14.   ],
+           [  14.875,   75.125,   14.   ],
+           [ 155.125,  105.875,   12.   ],
+           [  92.75 ,   46.625,   14.   ],
+           [ 159.625,  127.25 ,   18.25 ],
+           [ 107.625,   16.5  ,   12.25 ],
+           [   0.875,  191.875,   14.5  ],
+           [  45.625,  142.125,   12.5  ],
+           [ 137.875,  192.   ,   12.25 ],
+           [ 106.625,   59.125,   13.   ],
+           [  91.375,   18.5  ,   13.25 ],
+           [ 152.   ,  184.125,   12.5  ],
+           [ 149.125,   91.125,   12.   ],
+           [ 122.75 ,   28.875,   12.75 ],
+           [  82.625,  120.   ,   13.75 ],
+           [  95.875,   30.75 ,   12.25 ],
+           [  16.125,  165.25 ,   15.75 ],
+           [  57.875,   76.375,   23.5  ],
+           [  86.125,   56.5  ,   12.   ],
+           [  80.75 ,   29.875,   12.5  ]])
+
+    Notes
+    -----
+    The radius of each region is 'scale/2'.
+    """
+
+    assert_nD(image, 2)
+
+    # scales for keypoints
+    scales = np.array([_ for _ in range(min_scale, max_scale, 2)])
+    # detect keypoints in scale-space
+    base_regions = detect(image, scales)
+    # pruning based on thresholds
+    regions = prune(base_regions, saliency_threshold, clustering_threshold, K=7)
+
+    return regions.T
+
+
+def detect(image, scales):
+    """ Calculates the Shannon entropy of local image attributes for each region
+    over a range of scales, and weight them based on the difference between
+    the local descriptors around scale-space.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input grayscale image
+    scales : ndarray
+        Scales for key regions
+    Returns
+    -------
+    A : (4, n) ndarray
+        A 2d array with each column representing 4 values, '(gamma, scale, y, x)'
+        where '(y, x)' are coordinates of the region, 'scale' is the
+        size of salient region, and 'gamma' is the saliency score.
+    """
+    image = img_as_ubyte(image)
+    nr, nc = image.shape
+    # find pixels that we are going to examine
+    mask = np.ones((nr, nc))
+    r, c = np.nonzero(mask)
+    nPix = len(r)
+
+    nScales = len(scales)
+    intensity_edges = [_ for _ in range(0, 257, 16)]
+
+    previous_h = np.zeros((len(intensity_edges)-1, nPix))
+    weights = np.zeros((nScales, nPix))
+    entropy = np.zeros((nScales, nPix))
+
+    # iterate through every possible region
+    # iterate through scales
+    for s_count in range(nScales):
+        scale_size = scales[s_count]
+        radius = int((scale_size)/2)
+
+        # iterate through pixels
+        for i in range(nPix):
```

Comment:
This will be very slow, but, more importantly, can be made much simpler by using `skimage.util.view_as_windows`!